osp/core/session/session.py

```diff
@@ -18,13 +18,19 @@ def __init__(self):
         self._registry = Registry()
         self.root = None
         self.graph = rdflib.Graph()
+        self._previous_session = None
 
     def __enter__(self):
         """Establish the connection to the backend."""
+        from osp.core.cuds import Cuds
```

Comment:
It makes not difference in practice, but would it not make sense to just import Cuds once at the top?

Reply:
Unfortunately, it sometimes happens that there are circular imports. I tried to put it at the top, got such an error and moved it there. Ideally, we should structure our imports s.t. no circular imports occur. I didn't check why it occured in this case.